cmd/entrypoint/runner_test.go

```diff
@@ -14,9 +16,23 @@ func TestRealRunnerSignalForwarding(t *testing.T) {
 	rr := realRunner{}
 	rr.signals = make(chan os.Signal, 1)
 	rr.signals <- syscall.SIGINT
-	if err := rr.Run("sleep", "3600"); err.Error() == "signal: interrupt" {
+	if err := rr.Run(context.Background(), "sleep", "3600"); err.Error() == "signal: interrupt" {
 		t.Logf("SIGINT forwarded to Entrypoint")
 	} else {
 		t.Fatalf("Unexpected error received: %v", err)
 	}
 }
+
+// TestRealRunnerTimeout tests whether cmd is killed after time.Second even though it's supposed to sleep for an hour.
+func TestRealRunnerTimeout(t *testing.T) {
+	rr := realRunner{}
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+	// For seconds and higher time units, err.Error() returns "signal: killed".
+	// For microseconds and lower time units, err.Error() returns "context deadline exceeded".
+	if err := rr.Run(ctx, "sleep", "3600"); err.Error() == "signal: killed" || err.Error() == "context deadline exceeded" {
```

Comment:
Could we change this to "sleep 10" or something similarly short? This would cause it to fail faster in case timeout logic is broken by a future change.

Reply:
the shortest you can get away with plz!

Reply:
also: why are there two different possibilities for the error message?

i think it might be nicer to catch this error and return our own, e.g. something like "killed process since deadline %d exceeded"

Reply:
I have changed it so that the process spawned by a command is now supposed to sleep for 10 milliseconds while the execution time deadline is set to 1 milliseconds.

As a sidenote, it seems like the context deadline is accurate up to about 3 milliseconds.
I.e., if the process is supposed to sleep for 1 second, a 997 millisecond deadline may or may not cause a timeout.

I'll change the error handling.
What happens is that if a context deadline is exceeded, the running process is killed which is why `Run()` would return `"signal: killed"` as an error.
I think that if the deadline was super short (i.e. microseconds or lower), the process did not even execute yet which is why `"context deadline exceeded"` was returned.

I now changed it such that if `Run()` returns an error, first the context error is checked to find out whether the context deadline has been exceeded. If the context error is nil, the error returned by Run() will be displayed.